### PR TITLE
Refacto story component

### PR DESCRIPTION
**Pull request by dtrucs**

### CODE REVIEW

File: src/modules/Visualizer/Story/Story.js [hunk 1]
```diff
@@ -20,123 +21,108 @@ const resetLayers = map => ({ layers = [] }) => {
...
-  state = {
-    step: 0,
-  }
+export const Story = ({ map, story: { beforeEach = [], slides }, setLegends, translate }) => {
+  const [step, setStep] = useState(0);
 
-  componentDidMount () {
-    this.initMap();
-  }
+  const slidesLength = slides.length - 1;
```
Comment (mabhub) [inline src/modules/Visualizer/Story/Story.js:27]:
```suggestion
  const slidesLength = React.useMemo(() => (slides.length - 1), [slides.length]);
```

File: src/modules/Visualizer/Story/Story.js [hunk 2]
```diff
@@ -20,123 +21,108 @@ const resetLayers = map => ({ layers = [] }) => {
...
+    setStep(prevStep => (prevStep === slidesLength
       ? 0
-      : prevStep + 1 }));
-  }
+      : prevStep + 1));
+  }, [slidesLength]);
 
-  initMap () {
-    const { map } = this.props;
+  const { title, content } = slides[step];
```
Comment (mabhub) [inline src/modules/Visualizer/Story/Story.js:47]:
```suggestion
  const { title, content } = React.useMemo(() => slides[step], [slides, step]);
```

Comment (mabhub) [reply src/modules/Visualizer/Story/Story.js:47]: I'm not really sure if this one is legit or not…

File: src/modules/Visualizer/Story/Story.js (hunk 1 above)
Comment (dtrucs) [reply src/modules/Visualizer/Story/Story.js:27]: With your suggestion, we allocate another function with `slides.length` as argument to return... `slides.length - 1`.
It's not a huge calculation and i am not sure if it's better for performance purposes.
What do you think @jrmi ?

Comment (mabhub) [reply src/modules/Visualizer/Story/Story.js:27]: This insure that `slidesLength` won't change between renders when slides do not change…

But ultimately, as it is just a Number, it does not really matter.

So yep, you can forget about that ;)

Comment (jrmi) [reply src/modules/Visualizer/Story/Story.js:27]: No need useMemo here, a number don't change is reference ! The useMemo is only usefull for high computation which is no the case here.

File: src/modules/Visualizer/Story/Story.js (hunk 2 above)
Comment (jrmi) [reply src/modules/Visualizer/Story/Story.js:47]: No high computation here also so don't need.

File: src/modules/Visualizer/Story/Story.js (hunk 1 above)
Comment (jrmi) [reply src/modules/Visualizer/Story/Story.js:27]: My problem is more the naming here. Are we really talking about length of a silde ? I propose `slideCount` instead.
```suggestion
  const slideCount = slides.length - 1;
```

**Pull request merged**